Add document about versioned merge engine (#523)
alibaba/fluss: website/docs/table-design/table-types/pk-table/merge-engines/versioned.md

# Versioned Merge Engine

> By specifying `'table.merge-engine' = 'versioned'`, users can update data based on the configured version column. Updates will be carried out when the latest value of the specified field is greater than the stored value. If it is less than or empty, no update will be made.

Reviewer: nit:
```suggestion
By setting `'table.merge-engine' = 'versioned'`, users can update data based on the configured version column. Updates will be carried out when the latest value of the specified field is greater than or equal to the stored value. If it is less than or null, no update will be made.
```

> 

Reviewer: I'd like to suggest to add a statement to explain how the feature is useful just like `first-row` merge page does, maybe something like:

```
This feature is particularly valuable for replacing [Deduplication](https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/dev/table/sql/queries/deduplication/) transformations in streaming computations, reducing complexity and improving overall efficiency.
```

:::note
When using `versioned` merge engine, there are the following limits:
- `UPDATE` and `DELETE` statements are not supported

> - Partial update is not supported

Reviewer: Add one more note just like first-row merge engine:
```
`UPDATE_BEFORE` and `DELETE` changelog events are ignored automatically
```

>   :::

Reviewer: ```suggestion
 :::
```

## Versioned Merge Column

:::note
The versioned merge column supports the following data types.
- INT
- BIGINT
- TIMESTAMP
- TIMESTAMP(p)
- TIMESTAMP_LTZ
- TIMESTAMP_LTZ(p)

>   :::

Reviewer: ```suggestion
 :::
```

example:
```sql title="versioned"

> create table merge_engine_with_version (

Reviewer: Could you please change SQL keyword to uppercase to align the style of first-row merge engine page?

    a int not null primary key not enforced,
    b string, 
    ts bigint
 ) with(
    'table.merge-engine' = 'versioned',
    'table.merge-engine.versioned.ver-column' = 'ts'
);

> insert into merge_engine_with_version ( a,b,ts ) VALUES (1, 'v1', 1000);

Reviewer: Haven't noticed that

-- insert data with ts < 1000, no update will be made
insert into merge_engine_with_version ( a,b,ts ) VALUES (1, 'v2', 999);
select * from merge_engine_with_version;
-- Output
-- +---+-----+------+
-- | a | b   | ts   |
-- +---+-----+------+
-- | 1 | v1  | 1000 |
-- +---+-----+------+

-- insert data with ts > 1000, update will be made
insert into merge_engine_with_version ( a,b,ts ) VALUES (1, 'v2', 2000);
select * from merge_engine_with_version;
-- Output
-- +---+-----+------+
-- | a | b   | ts   |
-- +---+-----+------+
-- | 1 | v2  | 2000 |

> 

Reviewer 2: Maybe add `-- +---+-----+------+` in table end.

-- insert data with ts = null, no update will be made

> nsert into merge_engine_with_version ( a,b,ts ) VALUES (1, 'v2', null);

Reviewer 2: typo: `nsert` ->  `insert`.

select * from merge_engine_with_version;
-- Output
-- +---+-----+------+
-- | a | b   | ts   |
-- +---+-----+------+
-- | 1 | v2  | 2000 |

> 

Reviewer 2: same as above.
